### ISSUE TITLE
skripts in command blocks?

### ISSUE THREAD

**Issue by SlimeDog** [SkriptLang/Skript#1511]
Spigot 1.13.1
Skript 2.3-alpha2

Most plugin commands can be executed in command blocks, but I cannot get a script to run. Is it possible?

**Comment by LeotomasMC** [comment]
Scripts arent commands. Scripts can create commands though. You should be able to use commands made by scripts in command blocks.

Note that @ selectors may not work with plugin commands in command blocks.

**Comment by Snow-Pyon** [comment]
does it work if you prefix the command with `skript;`?

**Comment by Blueyescat** [comment]
Already command blocks can use Skript commands, they can also run [effect commands](https://github.com/SkriptLang/Skript/blob/master/src/main/resources/config.sk#L64).

**Comment by SlimeDog** [comment]
The Skript command is:
```
command /starve:
        description: Starve player
        permission: blackdog.starve
        # permission message: &4You don't have permission to run this command!
        usage: /starve
        trigger:
...
```
I am able to insert
> skript:skript starve

into a command block. But nothing happens if I trigger it. (To verify that the command block fires, I placed another after it in the chain; the second block fires.) If I run the command manually, it works as expected.

**Comment by Blueyescat** [comment]
Any command that works in the server console will work on command blocks, that script also works for me. Adding `skript:` doesn't change anything. Skript 2.2-dev37c Paper 1.12.2

**Comment by Snow-Pyon** [comment]
@Blueyescat this is about 1.13 though, if something changed on how commands are handled (something I doubt but who knows), we gotta fix it.

**Comment by Nicofisi** [comment]
Once somebody is on the issue, make sure that Skript commands still work with  `Bukkit.dispatchCommand` too

**Comment by LeotomasMC** [comment]
the way commands work was very clearly changed. I remember one of the shapshots, you could use vannila server commands (eg. /ban, /kick) in singleplayer worlds. 1.13 has so many changes to so many things, its insane

**Comment by Blueyescat** [comment]
@TheBentoBox But in single player, even `/reload` lol.

**Comment by TheBentoBox** [comment]
Reload was added in 1.12

**Comment by Blueyescat** [comment]
I was talking about this https://www.youtube.com/watch?v=UKsGQmkZvGU but now i tried it in 1.13.1 and didn't work 🤔

**Comment by TheBentoBox** [comment]
Mojang's on to us

**Comment by Blueyescat** [comment]
Works fine! 🎉

**Comment by SlimeDog** [comment]
Will do.

**Comment by SlimeDog** [comment]
I tried:
> skript:starve

It is recognized in the commandblock, tab-completion. It just doesn't execute.

**Comment by Nicofisi** [comment]
Code, please

**Comment by SlimeDog** [comment]
Command is
> skript:starve

Skript is
```
command /starve:
        description: Starve player
        usage: /starve
        trigger:
                set {_player} to the player

                subtract 0.5 from hunger of {_player}
                set {_hunger} to hunger of {_player}

                set {_message} to ""
                if {_player} is in world "world_lobby":
                        if {_hunger} is 9:
                                set {_message} to "Aren't you getting hungry? Eat some cake!"
                        if {_hunger} is 8:
                                set {_message} to "You don't like cake?"
                        if {_hunger} is 7:
                                set {_message} to "Cake is soooooooo good!"
                        if {_hunger} is 6:
                                set {_message} to "We're out of cholocate cake. Try the red velvet!"
                        if {_hunger} is 5:
                                set {_message} to "Cake is health food!"
                        if {_hunger} is 4:
                                set {_message} to "You'll be sorry if you don't eat some cake!"
                        if {_hunger} is 3:
                                set {_message} to "You're going to regret this!"
                        if {_hunger} is 2:
                                set {_message} to "Don't say I didn't warn you!"
                        if {_hunger} is 1:
                                set {_message} to "Eat some cake before you die!"
                        if {_hunger} is less than 1:
                                set {_message} to "Sorry about that!"
                        {_message} is not equal to ""
                        send message "%{_message}%" to {_player}
                        exit
```

I also tried a version with player set explicitly from arg 1, with the command
> skript:starve %s

The result was the same.

**Comment by TheBentoBox** [comment]
How is the command block supposed to know what player to use? Excluding a target selector will definitely not work and `%s` isn't a valid target selector. You're probably thinking of `@s` but even then that wouldn't work because that targets the person executing the command, which is again the command block who is not a player. You'd need to add a player argument to the command and then send an actual player or a valid player selector as the argument to the command within the command block.

**Comment by Snow-Pyon** [comment]
So, I tested it with the exact same command, and I know what's happening now. Commands do work, what doesn't work is selectors in commands (though, in this specific case @s wouldn't have worked since the one who executed the command is the command block and not a player), this is a rather different bug that was reported before too.

**Comment by TheBentoBox** [comment]
Yeah, `@s` is the wrong target selector there. That makes the command target the one executing it. You'd be trying to starve the command block. Now, if the target selectors aren't working anyway, that's a different story, but you need to understand which target selectors to use when in the first place to use them properly. It's not often you need @s in command blocks.

**Comment by SlimeDog** [comment]
OK, I understand the mistake. Thanks for guidance.
